tests/bluetooth/controller/ctrl_cte_req/src/main.c

```diff
@@ -616,6 +616,749 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
 				  "Free CTX buffers %d", ctx_buffers_free());
 }
 
+/* Tests related with PHY update procedure and CTE request procedure "collision" */
+
+#define PREFER_S2_CODING 0U
+#define HOST_INITIATED 1U
+#define PHY_UPDATE_INSTANT_DELTA 6U
+#define PHY_PREFER_ANY (PHY_1M | PHY_2M | PHY_CODED)
+/* Arbitrary value used for setting effective maximum number of TX/RX octets */
+#define PDU_MAX_TX_RX_OCTETS (PDU_DC_PAYLOAD_SIZE_MIN * 3)
+
+static void check_pref_phy_state(const struct ll_conn *conn, uint8_t phy_tx, uint8_t phy_rx)
+{
+	zassert_equal(conn->phy_pref_rx, phy_rx,
+		      "Preferred RX PHY mismatch %d (actual) != %d (expected)", conn->phy_pref_rx,
+		      phy_rx);
+	zassert_equal(conn->phy_pref_tx, phy_tx,
+		      "Preferred TX PHY mismatch %d (actual) != %d (expected)", conn->phy_pref_tx,
+		      phy_tx);
+}
+
+static void check_current_phy_state(const struct ll_conn *conn, uint8_t phy_tx, uint8_t flags,
+				    uint8_t phy_rx)
+{
+	zassert_equal(conn->lll.phy_rx, phy_rx,
+		      "Current RX PHY mismatch %d (actual) != %d (expected)", conn->lll.phy_rx,
+		      phy_rx);
+	zassert_equal(conn->lll.phy_tx, phy_tx,
+		      "Current TX PHY mismatch %d (actual) != %d (expected)", conn->lll.phy_tx,
+		      phy_tx);
+	zassert_equal(conn->lll.phy_rx, phy_rx,
+		      "Current Flags mismatch %d (actual) != %d (expected)", conn->lll.phy_flags,
+		      flags);
+}
+
+static bool is_instant_reached(struct ll_conn *conn, uint16_t instant)
+{
+	/* Check if instant is in the past.
+	 *
+	 * NOTE: If conn_event > instant then subtract operation will result in value greather than
+	 *       0x7FFF for uint16_t type. This is based on modulo 65536 math. The 0x7FFF is
+	 *       maximum positive difference between actual value of connection event counter and
+	 *       instant.
+	 */
+	return (uint16_t)(event_counter(conn) - instant) <= (uint16_t)0x7FFF;
```

Comment:
Unfortunately the first cast it also needed. It is caused by integer conversion rules (or integer promotion rules).
In this case we have `event_counter(conn) - instant`  the result of this operation is `int` (at least according to my knowledge). It would have been converted to other type if there is e.g. assignment: `uint16_t value = event_counter(conn) - instant;.

In our case we have: `(event_counter(conn) - instant) <= (uint16_t)0x7FFF` -> `int <= (uint16_t)`.
in this case the `uint16_t` is promoted to `int`, because rank of `int` is greater than rank of `short int` and rank of `short int` it the same as rank of `uint16_t`. At the end of the day, the result of the expression is always `true`.

Reply:
With previous version of the code, one that you suggest to get back, the same rules apply. This is why there is an `& 0xFFFF` in the expression. Similar thing is applied with casts to `uint16_t`.